convert2rhel/backup.py

```diff
@@ -0,0 +1,180 @@
+            return False
+
+        cmd_param = ["rpm", "-i"]
+        if replace:
+            cmd_param.append("--replacepkgs")
+
+        loggerinst.info("Installing packages:")
+        for pkg in pkgs_to_install:
+            loggerinst.info("\t%s" % pkg)
+
+        cmd = cmd_param + pkgs_to_install
+        output, ret_code = run_subprocess(cmd, print_output=False)
+        if ret_code != 0:
+            pkgs_as_str = " ".join(pkgs_to_install)
+            loggerinst.debug(output.strip())
+            if critical:
+                loggerinst.critical("Error: Couldn't install %s packages." % pkgs_as_str)
+
+            loggerinst.warning("Couldn't install %s packages." % pkgs_as_str)
+            return False
+
+        for path in pkgs_to_install:
+            nvra, _ = os.path.splitext(os.path.basename(path))
+            self.track_installed_pkg(nvra)
+
+        return True
+
+    def restore_pkgs(self):
+        """Restore system to the original state."""
+        self._remove_installed_pkgs()
+        remove_orphan_folders()
+        self._install_removed_pkgs()
+
+
+class RestorableFile(object):
+    def __init__(self, filepath):
+        self.filepath = filepath
+
+    def backup(self):
+        """Save current version of a file"""
+        loggerinst.info("Backing up %s." % self.filepath)
+        if os.path.isfile(self.filepath):
+            try:
+                loggerinst.debug("Copying %s to %s." % (self.filepath, BACKUP_DIR))
+                shutil.copy2(self.filepath, BACKUP_DIR)
+            except (OSError, IOError) as err:
+                # IOError for py2 and OSError for py3
+                loggerinst.critical("Error(%s): %s" % (err.errno, err.strerror))
+        else:
+            loggerinst.info("Can't find %s.", self.filepath)
+
+    def restore(self):
+        """Restore a previously backed up file"""
+        backup_filepath = os.path.join(BACKUP_DIR, os.path.basename(self.filepath))
+        loggerinst.task("Rollback: Restoring %s from backup" % self.filepath)
+
+        if not os.path.isfile(backup_filepath):
+            loggerinst.warning("%s hasn't been backed up" % self.filepath)
+            return
+        try:
+            shutil.copy2(backup_filepath, self.filepath)
+        except (OSError, IOError) as err:
+            # Do not call 'critical' which would halt the program. We are in
+            # a rollback phase now and we want to rollback as much as possible.
+            # IOError for py2 and OSError for py3
+            loggerinst.warning("Error(%s): %s" % (err.errno, err.strerror))
+            return
+        loggerinst.info("File %s restored" % self.filepath)
+
+
+class RestorablePackage(object):
+    def __init__(self, pkgname):
+        self.name = pkgname
+        self.path = None
+
+    def backup(self):
+        """Save version of RPM package"""
+        loggerinst.info("Backing up %s" % self.name)
+        if os.path.isdir(BACKUP_DIR):
+            reposdir = get_eus_repos_available()
```

Comment:
```suggestion
            reposdir = get_hardcoded_repofiles_dir()
```
This is a bit tricky because we want to use the hardcoded repos for the package backup purposes only in case of non-latest RHEL release. Because imagine you have an air-gapped system with set up to receive updates from an internal Satellite server. With this change the package backup would fail. Without it it wouldn't.

Right of the bat I can't think of a good solution. We need to do some brainstorming here.